internal/pod/imagepuller.go

```diff
@@ -107,13 +122,47 @@ func (ipi *imagePullerImpl) ListPullPods(ctx context.Context, micObj *kmmv1beta1
 func (ipi *imagePullerImpl) GetPullPodForImage(pods []v1.Pod, image string) *v1.Pod {
 
 	for i, pod := range pods {
-		if image == pod.Labels[imageLabelKey] {
+		if image == pod.Spec.Containers[0].Image {
 			return &pods[i]
 		}
 	}
 	return nil
 }
 
 func (ipi *imagePullerImpl) GetPullPodImage(pod v1.Pod) string {
```

Comment:
Do we really need this public method? It doesn't give us anything that we can have access to in a single like outside of this package.

Reply:
image puller hides the implementation of the pull pod from the calling layer (in this case MBSC). Why should MBSC care about the details of how to get the status from pod?

Reply:
We are not getting the status from the pod but the image from the spec. If you prefer to keep it as is I don't mind but you know how I feel about primitive getters/setters :)

Reply:
sorry, my mistake! It makes the code simplier in the mbsc controller